models/workload.go

```diff
@@ -127,6 +131,10 @@ func (workload *WorkloadListItem) ParseWorkload(w *Workload) {
 	workload.PodCount = len(w.Pods)
 	workload.AdditionalDetailSample = w.AdditionalDetailSample
 
+	if len(w.Pods) > 0 {
```

Comment:
I'd expect the annotation on the workload, no need to get it from the Pod.
Just place the annotation inside the `parseObjectMeta` and that's good.
Note, that all workload annotations manipulation are located there.

Reply:
I don't know, the advantage of checking at pod level is that, in the end, there's always a pod ; whereas deployments are not mandatory, pods can result from various kind of controllers.
Like istio annotations in general are taken from pods, so I thought we could follow the same pattern.

Reply:
(see how many istio annotations are read from pods: https://istio.io/latest/docs/reference/config/annotations/ )

Reply:
In fact I checked that yesterday thanks @jotak.

In my last commit I am filtering by kiali Annotations.

Reply:
Deployments scaled to 0 are possible, a corner case but possible.
For this particular code, the suggestion is to move it into the parseObjectMeta() method that is performing a similar operation.
In that case, it will cover all cases, Pod included.